File: Solarness.Model/Project.cs
using System;
using System.Collections.Generic;

namespace Solarness.Model;

public partial class Project
{
    public int ProjectId { get; set; }

    public string ProjectName { get; set; } = null!;

    public string? ProjectDescription { get; set; }

    public string? StreetAddress { get; set; }

    public string? City { get; set; }

    public decimal? Kw { get; set; }

    public decimal? ContractAmount { get; set; }

    public DateOnly? SiteInspectionDate { get; set; }

    public DateOnly? EngineeringSubmitDate { get; set; }

    public DateOnly? EngineeringReceivedDate { get; set; }

    public DateOnly? SaleDate { get; set; }

    public string? Significance { get; set; }

    public string? Urgency { get; set; }

    public string? PriorityLevel { get; set; }

    public int? StatusId { get; set; }

    public int? TeamId { get; set; }

    public int? UserId { get; set; }

    public int? HomeownerId { get; set; }

    public virtual ProjectStatus? Status { get; set; }

    public virtual Team? Team { get; set; }
}

File: UI/solarness_desktop_app/lib/models/Project/project.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:solarness_desktop_app/models/Team/team.dart';
import 'package:solarness_desktop_app/models/Status/projectstatus.dart';

part 'project.g.dart';

@JsonSerializable()
class Project {
  int? projectId;
  String? projectName;
  String? projectDescription;
  String? streetAddress;
  String? city;
  int? kw;
  double? contractAmount;
  DateTime? siteInspectionDate;
  DateTime? engineeringSubmitDate;
  DateTime? engineeringReceivedDate;
  DateTime? saleDate;
  String? significance;
  String? urgency;
  String? priorityLevel;
  int? statusId;
  int? teamId;
  Team? team; 
  ProjectStatus? status;


  Project(this.projectId, this.projectName, this.projectDescription, this.streetAddress, this.city, this.kw,this.siteInspectionDate,this.engineeringSubmitDate,this.engineeringReceivedDate,
  this.saleDate,this.significance,this.urgency,this.priorityLevel,this.statusId,this.teamId, this.status,this.team,this.contractAmount);

  factory Project.fromJson(Map<String, dynamic> json) =>
      _$ProjectFromJson(json);

  Map<String, dynamic> toJson() => _$ProjectToJson(this);
}
